frontend/src/components/byte-parser/byte-parser.jsx

```diff
@@ -0,0 +1,342 @@
+import React, { Fragment, useState } from 'react';
+import { useContext } from 'use-context-selector';
+import { store } from 'store';
+
+import { Box, Node } from 'components';
+import { NODE } from 'utils/constants';
+
+import { Uint8, DATA_TYPES_BY_LENGTH } from './helpers';
+
+const toHexString = (byte) => byte.toString(16).padStart(2, '0').toUpperCase();
+
+const ROW_HEIGHT_PX = 16;
+const BYTES_IN_ROW = 8;
+const PADDING_TD = -1
+
+const ByteParser = () => {
+    const globalState = useContext(store);
+    const { dispatch, state } = globalState;
+
+    const TEST_DATA = Array.from(Uint8Array.from([65, 66, 67, 68, 0, 1, 2, 3, 0xff, 0xfe, 0xfd, 0xfc, 10, 0, 1, 2, 3, 0xff, 0xfe, 0xfd, 0xfc, 10]));
+    const BUFFER = Uint8Array.from([65, 66, 67, 68, 0, 1, 2, 3, 0xff, 0xfe, 0xfd, 0xfc, 10, 0, 1, 2, 3, 0xff, 0xfe, 0xfd, 0xfc, 10]);
+
+    const [startingSquare, setStartingSquare] = useState(null);
+    const [endingSquare, setEndingSquare] = useState(null);
+
+    const { groups } = state;
+
+    const _crossesGroup = (groups, startingSquare) => (endingSquare) => {

```

Comment:
I think the currying of a lot of these functions is unnecessary and hinders readability.  I would suggest refactoring to a single function if possible -- like for instance here, I think `endingSquare` could just be the argument to the function and `startingSquare` and `groups` don't need to be passed in at all since it has access to those in the component scope in the state they are in when the function is called.